### PR TITLE
added favouriteImage plugin

### PR DESCRIPTION
**Pull request by veygax**
This plugin allows you to favourite images instead of converting them to gifs.

### CODE REVIEW

**Comment by Golonchy** [comment]
waited a long time for this, 
epic

**Comment by MrDiamondDog** [comment]
Dupe of #2091

File: src/plugins/favouriteImage/index.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+
+export default definePlugin({
+    name: "FavouriteImage",
+    description: "Allows you to favourite an image.",
+    authors: [Devs.VeygaX,
+    {
```
Comment (MrDiamondDog) [inline src/plugins/favouriteImage/index.ts:14]: All authors should be in devs

File: src/plugins/favouriteImage/index.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        id: 457579346282938368n,
+        name: "Davri",
+    },
+    ],
+    start() {
+        RegExp._test ??= RegExp.prototype.test;
```
Comment (MrDiamondDog) [inline src/plugins/favouriteImage/index.ts:20]: Use patches instead of monkey patching RegExp.test

**Review by MrDiamondDog — COMMENTED** [review]
Just leaving this review as well

**Comment by happyendermangit** [comment]
Nice plugin, i waited a very very long for this!! the code is also clean.

File: src/plugins/favouriteImage/index.ts (hunk 2 above)
Comment (veygax) [reply src/plugins/favouriteImage/index.ts:15]: cén fá? ("why" in irish)

Comment (veygax) [reply src/plugins/favouriteImage/index.ts:15]: its works

**Comment by veygax** [comment]
> Dupe of #2091

that plugin is bloated + reports of slowing down discord 💀 my plugin just works

**Comment by sunnniee** [comment]
ignoring the fact that this plugin is [blatantly stolen](https://canary.discord.com/channels/603970300668805120/1117414445036081152/1118587629777195009), this is a horrible way of doing this, way worse than the "bloated" plugin made before this PR.
please don't come back again

**Comment by veygax** [comment]
> ignoring the fact that this plugin is [blatantly stolen]
I would have asked the original creator of the plugin for permission if he had been on in the past year bruh.
Also I credited the original snippet maker and my only intention was to make open a pr for the plugin.
‌
‌
‌
_And get a badge along the way 👄_

**Comment by veygax** [comment]
> this is a horrible way of doing this, way worse than the "bloated" plugin made before this PR.

what do you mean this is a horrible way of doing this? horrible to the undescribed standards that you guys set? 

it works and it does it without having 156 lines of code + a css file and modifying a discord feature which would cause the plugin to break if discord ever wants to update that feature bruh. 💀 at least mine is foolproof (kinda).

**Comment by happyendermangit** [comment]
This should be merged ASAP

**Comment by Nuckyz** [comment]
> > this is a horrible way of doing this, way worse than the "bloated" plugin made before this PR.
> 
> what do you mean this is a horrible way of doing this? horrible to the undescribed standards that you guys set?
> 
> it works and it does it without having 156 lines of code + a css file and modifying a discord feature which would cause the plugin to break if discord ever wants to update that feature bruh. 💀 at least mine is foolproof (kinda).

you could do the same with a single regex patch and it would be as bullet proof as this lmao

**Comment by veygax** [comment]
> you could do the same with a single regex patch and it would be as bullet proof as this lmao

Yeah but why would i take the hard way out

**Comment by Vendicated** [comment]
it's not the "hard way", it's the proper, clean and performant way

if you want to contribute, follow the project's style

**Pull request closed without merging**